src/components/StepperDialog/StepperDialog.spec.jsx

```diff
@@ -0,0 +1,96 @@
+)
+
+jest.mock('cozy-ui/transpiled/react/Dialog', () => ({
+  ...jest.requireActual('cozy-ui/transpiled/react/Dialog'),
+  __esModule: true,
+  default: 'MUIDialog'
+}))
+
+jest.mock('cozy-ui/transpiled/react/CozyDialogs', () => ({
+  ...jest.requireActual('cozy-ui/transpiled/react/CozyDialogs'),
+  DialogBackButton: ({ onClick }) => (
+    <div data-testid="DialogBackButton" onClick={onClick} />
+  ),
+  DialogCloseButton: ({ onClick }) => (
+    <div data-testid="DialogCloseButton" onClick={onClick} />
+  )
+}))
+/* eslint-enable react/display-name */
+
+describe('StepperDialog', () => {
+  it('should display "stepper" content', () => {
+    const { getByText } = render(<StepperDialog stepper={'1/2'} />)
+
+    expect(getByText('1/2')).toBeTruthy()
+  })
+
+  describe('DialogCloseButton', () => {
+    const closeAction = jest.fn()
+    it('should render DialogCloseButton in Desktop view & "onClose" prop is defined', () => {
+      useBreakpoints.mockReturnValue({ isMobile: false })
+      const { queryByTestId } = render(<StepperDialog onClose={closeAction} />)
+
+      expect(queryByTestId('DialogCloseButton')).toBeTruthy()
+    })
+
+    it('should not render DialogCloseButton in Desktop view & "onClose" prop is undefined', () => {
+      useBreakpoints.mockReturnValue({ isMobile: false })
+      const { queryByTestId } = render(<StepperDialog />)
+
+      expect(queryByTestId('DialogCloseButton')).not.toBeTruthy()
+    })
+
+    it('should not render DialogCloseButton in Mobile view even if "onClose" prop is defined', () => {
+      useBreakpoints.mockReturnValue({ isMobile: true })
+      const { queryByTestId } = render(<StepperDialog onClose={closeAction} />)
+
+      expect(queryByTestId('DialogCloseButton')).not.toBeTruthy()
+    })
+
+    it('should called closeAction function', () => {
+      useBreakpoints.mockReturnValue({ isMobile: false })
+      const { queryByTestId } = render(<StepperDialog onClose={closeAction} />)
+
+      const dialogCloseButton = queryByTestId('DialogCloseButton')
+
+      fireEvent.click(dialogCloseButton)
+
+      expect(closeAction).toBeCalledTimes(1)
+    })
+  })
+
+  describe('DialogBackButton', () => {
+    const backAction = jest.fn()
+    it('should render DialogBackButton if "onBack" prop is defined', () => {
+      const { queryByTestId } = render(<StepperDialog onBack={backAction} />)
+
+      expect(queryByTestId('DialogBackButton')).toBeTruthy()
+    })
+
+    it('should not render DialogBackButton if "onBack" prop is undefined', () => {
+      const { queryByTestId } = render(<StepperDialog />)
+
+      expect(queryByTestId('DialogBackButton')).not.toBeTruthy()
+    })
+
+    it('should called backAction function', () => {
+      const { queryByTestId } = render(<StepperDialog onBack={backAction} />)
+
+      const dialogBackButton = queryByTestId('DialogBackButton')
+
+      fireEvent.click(dialogBackButton)
+
+      expect(backAction).toBeCalledTimes(1)
+    })
+  })
+})
```

Comment:
le commit devrait plutôt être taggué `test: Add tests for StepperDialog component`
d'ailleurs on met plutôt les tests avec les features (je crois qu'on a ça quelque part dans les guidelines), je propose donc de squasher les 2

Reply:
Nein, pas de guidelines là-dessus.

Reply:
ok y'a donc pas mal de chose qu'on s'est dite qu'on n'a jamais écrit visiblement :)

Reply:
J'ai squash les 2 👍